### PR TITLE
Adds a devcontainer primarily for GitHub Codespaces use

### PR DESCRIPTION
**Pull request by colindean**
This will enable faster configuration of a development environment.

This is based off of https://github.com/Homebrew/brew/tree/4.1.14/.devcontainer.

Actively tested in GitHub Codespaces: I'm able to start the new Codespace, run `rspec`, and use git with editor integration as one expects.

### CODE REVIEW

File: .devcontainer/on-create-command.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+then
+  echo "Skipping slow items, not prebuilding."
+  exit 0
+fi
+
+# install Homebrew's development gems
+brew install-bundler-gems --groups=all
+
+# install Homebrew formulae we might need
+# brew install shellcheck shfmt gh gnu-tar
```
Comment (colindean) [inline .devcontainer/on-create-command.sh:19]: Could restore this:

```suggestion
brew install shellcheck shfmt gh gnu-tar
```

**Review by jacobbednarz — APPROVED** [review]

File: .devcontainer/on-create-command.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+# cleanup any mess
+brew cleanup
+
+# actually tap homebrew/core, no longer done by default
+# brew tap homebrew/core
+
```
Comment (MikeMcQuaid) [inline .devcontainer/on-create-command.sh:26]:
```suggestion
```

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    "workspaceMount": "source=${localWorkspaceFolder},target=/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-bundle/,type=bind,consistency=cached",
+    "onCreateCommand": "./.devcontainer/on-create-command.sh",
+    "postCreateCommand": "bundle install",
+    "remoteEnv": {
+      "HOMEBREW_GITHUB_API_TOKEN": "${localEnv:GITHUB_TOKEN}",
+      "PATH": "${containerEnv:PATH}:/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/current/bin",
+      "EDITOR": "code --wait",
+      "VISUAL": "code --wait"
+    }
+  }
```
Comment (MikeMcQuaid) [inline .devcontainer/devcontainer.json:14]:
```suggestion
  }

```

**Review by MikeMcQuaid — APPROVED** [review]
Thanks @colindean!

**Comment by colindean** [comment]
Would you mind adding the `hacktoberfest-accepted` label to this PR or opting into [Hacktoberfest](https://www.hacktoberfest.com) by adding the `hacktoberfest` topic to this repo using the gear menu on the repo page next to the About box?

**Comment by MikeMcQuaid** [comment]
@colindean have done the prior!

**Comment by colindean** [comment]
Thank you!

**Pull request merged**